src/databricks/labs/ucx/progress/tables.py

```diff
@@ -38,17 +45,24 @@ def __init__(
             schema,
             table,
         )
-        self._table_migration_index = table_migration_index
+        self._migration_status_refresher = migration_status_refresher
+
+    def append_inventory_snapshot(self, snapshot: Iterable[Table]) -> None:
+        migration_index = TableMigrationIndex(self._migration_status_refresher.snapshot())
```

Comment:
We commonly use a `cached_property` for this "caching" of overwriting the `append_inventory_snapshot1

Reply:
That would be wrong, because the snapshot has to correspond to the snapshot that is passed in. The caller is responsible for controlling when refreshes occur, and caching this internally would defeat the control of the caller being able to control that.